### ISSUE TITLE
Intelligence has a cost

### ISSUE THREAD

**Issue by thedaveCA** [davidrmiller/biosim4#4]
One thought about the number of neurons and connections, it could be interesting to randomize the genome length and inner neurons, but introduce a cost associated with each.

The cost could be implemented as requiring a rest between movements (cycles where an individual is simply skipped and remains in place).

To see the effects it might make sense to increase the size of the grid, and gradually reduce the number of steps/generation.

Smaller brains will probably start to win out even before reducing the number of steps as they will be more likely to hit the breeding area first, leaving the smarter ones to possibly arrive in a breeding area that is already full, but moving the breeding targets (perhaps on a wall, but not a corner) could provide a motivation. Barriers, in particular those that form a corner that an individual would need to overcome could be interesting as this would start to require "distance from wall" decisions.

There is a lot more here than I would be capable of coding, so consider this a feature request if it makes sense?

**Comment by Infiniti20** [comment]
Yeah that's definitely an interesting thought. Once I'm finished my current project, I might work on making this simulation a little more complex, like adding food, water, and a whole host of things.

**Comment by davidrmiller** [comment]
Those are interesting ideas and well worth exploring. At one point I tried to allow genomes to become longer or shorter by mutation, but found that the genomes tended to infinitely expand or contract. My time was limited, so I abandoned those experiments, but there is a lot to explore there. I also tried adding an "energy" property to the individuals that got depleted by activity (which is somewhat proportional to genome length) and automatically elevated by inactivity. The results were interesting but unruly, and I didn't pursue those experiments either.
